util/types/convert.go

```diff
@@ -257,11 +262,44 @@ func StrToFloat(str string) (float64, error) {
 	if len(str) == 0 {
 		return 0, nil
 	}
+	validStr := getValidFloatPrefix(str)
```

Comment:
Warnings will be implemented in the future.
The function itself just return errors, if the statement is not `insert` or `update`, or sql_mode is not strict, errors will be converted to warnings.